storage.go

```diff
@@ -530,6 +469,141 @@ func (s *Storage) nextPartPage(ctx context.Context, page *PartPage) error {
 	return nil
 }
 
+func (s *Storage) querySignHTTPCompleteMultipart(ctx context.Context, o *Object, parts []*Part, expire time.Duration, opt pairStorageQuerySignHTTPCompleteMultipart) (req *http.Request, err error) {
+	pairs, err := s.parsePairStorageCompleteMultipart(opt.pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	input := s.formatCompleteMultipartUploadInput(o, parts, pairs)
+
+	completeReq, _ := s.service.CompleteMultipartUploadRequest(input)
+	url, headers, err := completeReq.PresignRequest(expire)
+	if err != nil {
+		return nil, err
+	}
+
+	var buf bytes.Buffer
+	err = xmlutil.BuildXML(completeReq.Params, xml.NewEncoder(&buf))
+	if err != nil {
+		return nil, err
+	}
+
+	req, err = http.NewRequest("POST", url, &buf)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header = headers
+	return
+}
+
+func (s *Storage) querySignHTTPCreateMultipart(ctx context.Context, path string, expire time.Duration, opt pairStorageQuerySignHTTPCreateMultipart) (req *http.Request, err error) {
+	pairs, err := s.parsePairStorageCreateMultipart(opt.pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	input, err := s.formatCreateMultipartUploadInput(path, pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	multipartReq, _ := s.service.CreateMultipartUploadRequest(input)
+	url, headers, err := multipartReq.PresignRequest(expire)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err = http.NewRequest("POST", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header = headers
+	return
+}
+
+func (s *Storage) querySignHTTPDelete(ctx context.Context, path string, expire time.Duration, opt pairStorageQuerySignHTTPDelete) (req *http.Request, err error) {
+	pairs, err := s.parsePairStorageDelete(opt.pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	if pairs.HasMultipartID {
```

Comment:
Maybe we should also support `ObjectMode` to delete dir?

Reply:
We must use `AbortMultipart` to remove the `multipart object` before `completeMultipart`, in this case we must use `multipartID`.

Reply:
Maybe you're right.

Reply:
Oh, I added it, in [utils.go](https://github.com/beyondstorage/go-service-s3/pull/237/files?authenticity_token=mdOfA5eA5gsgNOoesXvE8DFlLiMmL9vWIxv9khbU8ap5PeWudO7VPWg%2F4QXljW0kdUDcR6wSfugQW1rqKBuAtg%3D%3D&file-filters%5B%5D=.go&file-filters%5B%5D=.mod#diff-8b7f25bac2ae04c2ce2abb628c6aa28103fd113103a62f3e9d232fa995684915R447).

Reply:
Maybe I shouldn't put it in `utils.go`.

Reply:
Oh, I missed it.